source/http/HttpConnection.cpp

```diff
@@ -168,41 +163,38 @@ namespace Aws
 
                 /* Do the same ref counting trick we did with HttpClientConnection. We need to maintain a reference
                  * internally (regardless of what the user does), until the Stream shuts down. */
-                auto *toSeat = static_cast<HttpClientStream *>(aws_mem_acquire(m_allocator, sizeof(HttpStream)));
+                auto *toSeat = static_cast<HttpClientStream *>(aws_mem_acquire(m_allocator, sizeof(HttpClientStream)));
```

Comment:
duude. I found exactly this kind of bug in my own code today too... sizeof'ing the wrong thing but it happened to be big enough anyway